init.c

```diff
@@ -292,11 +279,9 @@ setup_smoothinglengths(int RestartSnapNum)
         myfree(olddensity);
 #endif //DENSITY_INDEPENDENT_SPH
 #pragma omp parallel for
-        for(i = 0; i < NumPart; i++) {
-            if(P[i].Type == 0) {
-                /* EgyWtDensity stabilized, now we convert from energy to entropy*/
-                SPHP(i).Entropy = GAMMA_MINUS1 * u_init / pow(SPHP(i).EOMDensity/a3 , GAMMA_MINUS1);
-            }
+        for(i = 0; i < N_sph_slots; i++) {
```

Comment:
I am not sure -- there are cons and pros directly accessing SphP -- This way if N_sph < N_sph_slots you may end up having some nans in SphP[i]?

Reply:
At init this is ok, I think. I can check for EOMDensity == 0 if you like.

Reply:
I slightly prefer to access SphP via the accessor than directly meddling with the underlying storage.

It is slower but the API is more consistent. We shall make garbage.c the only place where SphP / BhP is directly modified (and promote it to slotmanager.c).

If the skip by ptype bothers you, we may want to have the indirect sorting index by PType, like the one used in petaio -- the indirect sorting index did significant improved speed for preparing IO buffer of BH and STAR there.

Reply:
Ok, no worries, I will drop the commit.